nixos/modules/services/editors/emacs.xml

```diff
@@ -0,0 +1,561 @@
+        <link xlink:href="http://homepage.boetes.org/software/mg/">mg</link>,
+        <link xlink:href="http://yi-editor.github.io/">Yi</link>.
+      </para>
+
+    </section>
+    <section>
+      <title>Adding Packages to Emacs</title>
+      <para>
+        Emacs includes an entire ecosystem of functionality beyond
+        text editing, including a project planner, mail and news
+        reader, debugger interface, calendar, and more.
+      </para>
+
+      <para>
+        Most extensions are gotten with the Emacs packaging system
+        (<filename>package.el</filename>) from <link
+        xlink:href="https://elpa.gnu.org/">Emacs Lisp Package Archive
+        (<acronym>ELPA</acronym>)</link>,
+        <link xlink:href="https://melpa.org/"><acronym>MELPA</acronym></link>,
+        <link xlink:href="https://stable.melpa.org/">MELPA Stable</link>,
+        and <link xlink:href="http://orgmode.org/elpa.html">Org ELPA</link>.
+        Nixpkgs is regularly updated to mirror all these archives.
+      </para>
+
+      <para>
+        Under NixOS, you can continue to use
+        <function>package-list-packages</function> and
+        <function>package-install</function> to install packages. You
+        can also declare the set of Emacs packages you need using the
+        derivations from Nixpkgs. The rest of this section discusses
+        declarative installation of Emacs packages through nixpkgs.
+      </para>
+
+      <note>
+        <para>
+          This documentation describes the new Emacs packages
+          framework in NixOS 16.03
+          (<varname>emacsPackagesNg</varname>) which should not be
+          confused with the previous and deprecated framework
+          (<varname>emacs24Packages</varname>).
+        </para>
+      </note>
+
+      <para>
+        The first step to declare the list of packages you want in
+        your Emacs installation is to create a dedicated
+        derivation. This can be done in a dedicated
+        <filename>emacs.nix</filename> file such as:
+
+      <example xml:id="ex-emacsNix">
+        <title>Nix expression to build Emacs with packages (<filename>emacs.nix</filename>)</title>
+        <programlisting language="nix">
+/*
+This is a nix expression to build Emacs and some Emacs packages I like
+from source on any distribution where Nix is installed. This will install
+all the dependencies from the nixpkgs repository and build the binary files
+without interfering with the host distribution.
+
+To build the project, type the following from the current directory:
+
+$ nix-build emacs.nix
+
+To run the newly compiled executable:
+
+$ ./result/bin/emacs
+*/
+{ pkgs ? import &lt;nixpkgs&gt; {} }: <co xml:id="ex-emacsNix-1" />
+
+let
+  myEmacs = pkgs.emacs; <co xml:id="ex-emacsNix-2" />
+in
+(pkgs.emacsPackagesNgGen myEmacs).emacsWithPackages ((with pkgs.emacsPackagesNg.melpaStablePackages; [ <co xml:id="ex-emacsNix-3" />
```

Comment:
Instead of 
```(pkgs.emacsPackagesNgGen myEmacs).emacsWithPackages ((with pkgs.emacsPackagesNg.melpaStablePackages;
```
this should be
```
(pkgs.emacsPackagesNgGen myEmacs).emacsWithPackages (emacsPkgs: (with emacsPkgs; [
```
It's not obvious, but the way you have it, the packages will be built to use a different Emacs than `myEmacs`!